toxcore/tox.h

```diff
@@ -340,6 +340,24 @@ uint32_t tox_max_filename_length(void);
 
 uint32_t tox_max_hostname_length(void);
 
+/**
+ * Maximum length of a SOCKS5 proxy username in bytes.
+ *
+ * @deprecated The macro will be removed in 0.3.0. Use the function instead.
+ */
+#define TOX_MAX_PROXY_SOCKS5_USERNAME_LENGTH 255
```

Comment:
I know these used to be generated, but should we really add new deprecated-at-the-time-of-creation macros?

Reply:
I wonder.

We have also recently changed the API from using `*buffer` to `buffer[SIZE_CONSTANT]` in https://github.com/TokTok/c-toxcore/pull/2518, e.g.

```C
bool tox_bootstrap(Tox *tox, const char *host, uint16_t port, const uint8_t public_key[TOX_PUBLIC_KEY_SIZE], Tox_Err_Bootstrap *error);
```

Why are we making the API functions use deprecated macros?

Are we actually keeping the macros now, not deprecating them?

@iphydf ?

Anyway, I made the API changes such that they would match what the rest of the API does, that is: define constants and use them for array length in function declarations. You can see those used in the function declarations:

```C
void tox_options_set_proxy_socks5_username(Tox_Options *options, const uint8_t username[TOX_MAX_PROXY_SOCKS5_USERNAME_LENGTH], size_t length);

void tox_options_set_proxy_socks5_password(Tox_Options *options, const uint8_t password[TOX_MAX_PROXY_SOCKS5_PASSWORD_LENGTH], size_t length);
```

I think it makes sense to add the macros, even if they are deprecated at the creation, in order to keep the API coherent. Currently everything has macros. Once we decide to remove the macros, we can remove them all at once and nothing will have macros. That way there is no in-between state where some things have macros but some don't. (Please point out if we are already in the in-between state where some of the added APIs don't define macros, in which case I will remove the macros too. I checked that the ngc defines macros, so figured we are still in the all-macro state, but I might have missed something.)

Reply:
The arrays are not always of max size, or might not be arrays at all.

Reply:
@Green-Sky I don't understand what you are saying, can you re-phrase that or provide an example of what do you mean?

Reply:
If I am not mistaken, then the size is supposed to be exact, so you would have to provide a max sized buffer (if your code is also checked by cimple).

so something like
```
const char* socks_host = "localhost";
```
would be strongly discoraged.

Reply:
@Green-Sky If you are saying that an API user can pass `uint8_t *` for the array and that `uint8_t array[SIZE]` doesn't mean that the array the user passes is of exactly `SIZE` length (in fact, if you do `sizeof(array)` on a finction with `uint8_t array[SIZE]` argument, you will notice that it's equal to `sizeof(uint8_t *)`) -- then yes, that's how it indeed works. The `SIZE` in `uint8_t array[SIZE]` is merely for the documentation purposes.

Still, I don't understand what you meant by stating the obvious. Is there some issue in here? Do you want something in the PR to be changed? Please be more explicit!

Reply:
@Green-Sky Oh, your 2nd comment didn't show up for me until I refreshed the page just now.

It seems like you are thinking of `uint8_t array[static SIZE]` instead of `uint8_t array[SIZE]`. It's the `static` variant that promises for the array to be at least `SIZE` in length. The non-static variant doesn't make any promises, it's equivalent to `uint8_t array[]`. Please correct if I'm wrong though, it might be me who is confusing them.

Reply:
Ok, I think I understand what you mean. I just checked the API and we have the size specified only for arrays that should always be of that exact size. So it's a cimple thing, not a C standard thing?

Alright, I will be removing those size specifiers then, thanks for letting me know and bearing with my confusion, @Green-Sky.

Reply:
@Green-Sky is this better? You can see the diff as r2 vs r3 in Reviewable (assuming it shows the same revisions to everyone).

Reply:
@nurupo sorry, this noti got burried in my github inbox.

>  I just checked the API and we have the size specified only for arrays that should always be of that exact size. So it's a cimple thing, not a C standard thing?

Yea, that is what I meant. We should probably specify when we talk about the toxcore c-dialect cimple. :)